pkg/cfn/template/iam.go

```diff
@@ -0,0 +1,39 @@
+package template
+
+// IAMPolicy represents a CloudFormation AWS::IAM::Policy resource
+type IAMPolicy struct {
+	PolicyName *Value `json:",omitempty"`
+
+	Roles          []*Value        `json:",omitempty"`
+	PolicyDocument MapOfInterfaces `json:",omitempty"`
```

Comment:
We don't use references for policy documents, so we declare it as a map of interfaces, as that's how we always use it.